tests/by-util/test_tail.rs

```diff
@@ -2435,6 +2423,658 @@ fn test_illegal_seek() {
     );
     assert_eq!(p.wait().unwrap().code().unwrap(), 1);
 }
+// FIXME: Occasionally broken pipe on macos when no output is expected. Switch on such cases for macos when fixed.
+#[cfg(all(not(target_os = "android"), not(target_os = "windows")))] // FIXME: See https://github.com/uutils/coreutils/issues/3881
+mod pipe_tests {
+    use super::*;
+    use crate::common::random::*;
+    use rand::distributions::Alphanumeric;
+    const CHUNK_BUFFER_SIZE: usize = 8192;
+
+    #[test]
+    fn test_pipe_when_lines_option_value_is_higher_than_contained_lines() {
+        let test_string = "a\nb\n";
+        new_ucmd!()
+            .args(&["-n", "3"])
+            .pipe_in(test_string)
+            .succeeds()
+            .stdout_only(test_string);
+
+        new_ucmd!()
+            .args(&["-n", "4"])
+            .pipe_in(test_string)
+            .succeeds()
+            .stdout_only(test_string);
+
+        new_ucmd!()
+            .args(&["-n", "999"])
+            .pipe_in(test_string)
+            .succeeds()
+            .stdout_only(test_string);
+
+        #[cfg(all(not(target_os = "macos")))]
```

Comment:
I tried this (and the other cfgs) on a macOS system. If uncommented the test sometimes succeeds and sometimes fails. If it fails it runs into the `!todo()` at `tail.rs:434:17`.
It's very strange, that it fails only sometimes. I suspect it has something to do with the handling of the legacy `+` syntax.

Reply:
That's exaclty the same behaviour I observed when running the Github Actions CI jobs. The tests failed randomly on macos when no output was expected, at least this is what I suspected. Some tests also failed in cases of `-n -0` or `-c -0`.

Reply:
I did some more testing on macOS.
The failures will also sometimes (10-20% of the time) happen on the `main` branch.
However, they will only happen inside the test framework, i.e. this will never trigger a failure:
```bash
for i in {1..100}; do echo "a\nb\n" | target/debug/tail -n +3; done
```
but this will fail regularly:
```rust
let test_string = "a\nb\n";
new_ucmd!()
    .args(&["-n", "+3"])
    .pipe_in(test_string)
    .succeeds()
    .no_stdout()
    .no_stderr();
```

Reply:
Strange indeed. So you think it has something to do with the test system? It's maybe better to move this to an own issue.

Reply:
I've created an issue #3895 and added infos which might help narrowing down the error.